### ISSUE TITLE
`diff` keyword is not allowed to be defined with variable

### ISSUE THREAD

**Issue by Vanav** [ansible/ansible-lint#2771]
Source: `roles/base/tasks/main.yml`
```
- name: byobu status
  template:
    src: byobu-status
    dest: /root/.byobu/status
  diff: '{{ show_diff }}'
```

Result: `ansible-lint`

---
schema[tasks]: {'name': 'byobu status', 'template': {'src': 'byobu-status', 'dest': '/root/.byobu/status'}, 'diff': '{{ show_diff }}'} is not valid under any of the given schemas (warning)
roles/base/tasks/main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

Schema bugs should be reported towards [schemas](https://github.com/ansible/schemas) project instead of ansible-lint.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.
---

Seems that `diff` is not allowed to be defined with variable, but this syntax seems valid and working in Ansible (https://docs.ansible.com/ansible/latest/reference_appendices/playbooks_keywords.html).
